### PR TITLE
Rename the exporter (NEWRELIC-6322)

### PR DESCRIPTION
**Pull request by kang-makes**
We finally decided to keep two versions of the mongo integration at the same time, the old one that uses an outdated library and this one that leverages the MongoDB exporter.

As the exporters break everything, not only use a completely different codebase but also send a different set of metrics, y break also dashboards and alerts.

Because of that, we are renaming this to `mongodb3` so that in a package manager you can install `nri-mongodb` and `nri-mongodb3` at the same time.

### CODE REVIEW

**Comment by paologallinaharbur** [comment]
Notice that (unrelated to your PR) the pre-release of MongoDB failed since the e2e action is run as well at release time and we forgot to fix it there as well.

Can you fix that as well? Or we can open a different PR :)

**Comment by kang-makes** [comment]
Tests are now passing. Also fixed the issue @paologallinaharbur pointed me out as it is simple enough to not pollute this review.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -5,20 +5,16 @@ on:
...
     steps:
-      - name: checkout-repository
-        uses: actions/checkout@v2
-      - name: Setup go
-        uses: actions/setup-go@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v2
         with:
-          go-version: ${{env.GO_VERSION}}
+          go-version-file: nri-config-generator/go.mod
```
Comment (sigilioso) [inline .github/workflows/e2e.yml:17]: Thank you for addressing this as well 🙌

**Review by sigilioso — APPROVED** [review]
LGTM

**Pull request merged**